modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -785,6 +785,70 @@ public void registerObjectRelationshipsRelatedInfoCollectionProviders(
 		}
 	}
 
+	public ObjectRelationship updateEdge(
+			long objectRelationshipId, boolean edge)
+		throws PortalException {
+
+		ObjectRelationship objectRelationship =
+			objectRelationshipLocalService.getObjectRelationship(
+				objectRelationshipId);
+
+		if (!edge ||
+			!Objects.equals(
+				objectRelationship.getDeletionType(),
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE)) {
+
+			return objectRelationship;
+		}
+
+		ObjectDefinitionLocalService objectDefinitionLocalService =
+			_objectDefinitionLocalServiceSnapshot.get();
+
+		ObjectDefinition objectDefinition1 =
+			objectDefinitionLocalService.getObjectDefinition(
+				objectRelationship.getObjectDefinitionId1());
+		ObjectDefinition objectDefinition2 =
+			objectDefinitionLocalService.getObjectDefinition(
+				objectRelationship.getObjectDefinitionId2());
+
+		if (objectDefinition1.getStatus() != objectDefinition2.getStatus()) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					StringBundler.concat(
+						"Object relationship ID ", objectRelationshipId,
+						" cannot be  edge because the object definitions have ",
+						"different status"));
+			}
+
+			return objectRelationship;
+		}
+		else if (objectDefinition1.getRootObjectDefinitionId() !=
+					objectDefinition2.getRootObjectDefinitionId()) {
```

Comment:
@G-Sa can we first check if `objectDefinition1.getRootObjectDefinitionId() != objectDefinition2.getRootObjectDefinitionId()`? 

Alphabetical order :P